resource/environment/integration.js

```diff
@@ -95,169 +94,304 @@ function onEnvironment( data, env ) {
+		templateResult[response[i].name] = yield rp( response[i].download_url, {
+			qs: {
+				access_token: token // eslint-disable-line
+			},
+			headers: {
+				"User-Agent": "cowpoke"
+			}
+		} );
+	}
+	return templateResult;
+});
+
+const getTemplate = Promise.coroutine(function* ( token, catalogOwner, catalog, info ) {
+
+	let response =  yield rp( format( "https://api.github.com/repos/%s/%s/contents/templates/%s?ref=master", catalogOwner, catalog, info.branch ), {
+		qs: {
+			access_token: token // eslint-disable-line
+		},
+		headers: {
+			"User-Agent": "cowpoke"
+		},
+		json: true
+	} ).catch(err => {
+		console.error(err);
+		return undefined;
+	});
+
+	if (response === undefined) {
+		return undefined;
+	}
+	
+	const templateRequests = [];
+	for (let i = 0; i < response.length; i++) {
+		if ( isnum( response[i].name ) ) {
+			templateRequests.push(rp( response[i]._links.self, {
+				qs: {
+					access_token: token // eslint-disable-line
+				},
+				headers: {
+					"User-Agent": "cowpoke"
+				},
+				json: true
+			}).then(readTemplate.bind(null, token, response[i].name)));
+		}
+	}
+	let templates = yield Promise.all(templateRequests);
+	for (let i = 0; i < templates.length; i++) {
+		let dockerCompose = yaml.safeLoad( templates[i]["docker-compose.yml"] );
+		for (let service in dockerCompose) {
+			const currentImage = util.getImageInfo( dockerCompose[service].image );
+			if (dockerCompose.hasOwnProperty(service) && currentImage && currentImage.newImage === info.newImage) {
+				return  templates[i];
+			}
+		}
+	}
+	return undefined;
+});
+
+const upgradeStack = Promise.coroutine(function* ( slack, dockerhub, github, envelope ) {
+	const info = util.getImageInfo( envelope.data.docker_image );
+	if (!info) {
+		return invalidTagError(envelope.data.docker_image);
+	}
+	const exists = dockerhub.checkExistance( envelope.data.docker_image );
+	if (exists === undefined) {
+		return dockerError;
+	} else if (!exists) {
+		return tagNotFoundError;
+	}
+	//get the template
+	const template = yield getTemplate( github.token, github.owner, envelope.data.rancher_catalog, info);
+	if ( template === undefined ) { 
+		return {
+			status: 404,
+			data: {
+				message: "Unable to get information from github"
+			}
 		};
 	}
-	return environment.getByName( name ).then( onEnv );
-}
+	//get the environments
+	const storedEnvironments = yield environment.getAll().catch( () => undefined );
+	if ( storedEnvironments === undefined ) { 
+		return readError;
+	}
+	//get the rancher data
+	const envRequests = [];
+	for ( let i = 0; i < storedEnvironments.length; i++ ) {
+		envRequests.push( rancherFn( storedEnvironments[i].baseUrl, {
+			key: storedEnvironments[i].key,
+			secret: storedEnvironments[i].secret
+		} )
+		.then( rancher => rancher.listEnvironments())
+		.then(environments => environments[_.keys( environments )[0]])
+		.catch(error => {
+			console.log("error while trying to connect to " + storedEnviorments[i].name, ": ", error);
```

Comment:
@EvanSnapp - everywhere else, it's correct, but in the error message this will cause a reference error because of the typo, change to `storedEnvironments` :)